WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java

```diff
@@ -86,6 +86,12 @@ public void onPause() {
     @Override
     public void onResume() {
         super.onResume();
+
+        // Site name may have changed (e.g. via Settings and returning to this Fragment) so update the UI
+        if (mBlog != null && !mBlogTitleTextView.getText().equals(mBlog.getBlogName())) {
```

Comment:
> Hmm, indeed. I used this method:

Yuck. Seems that we should have a common routine which displays the blog name & url using the same technique as `refreshBlogDetails` (which appears to deal with the rare situation that the blog name is empty).